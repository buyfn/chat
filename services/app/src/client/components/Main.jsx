import React from 'react';

import MessagesContainer from '../containers/Messages';
import FormContainer from '../containers/Form';
import UserCounterContainer from '../containers/UserCounter';
import LoginFormContainer from '../containers/LoginForm';
import SignupFormContainer from '../containers/SignupForm';
import MenuContainer from '../containers/Menu';
import AuthForm from '../containers/AuthForm';

const Main = () => (
  <div>
    <UserCounterContainer />

    <MenuContainer />

    <AuthForm formType="login">
      <LoginFormContainer />
    </AuthForm>

    <AuthForm formType="signup">
      <SignupFormContainer />
    </AuthForm>

    <FormContainer />

    <MessagesContainer />
  </div>
);

export default Main;
