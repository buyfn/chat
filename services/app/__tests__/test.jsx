import '@babel/polyfill';

import React from 'react';
import { Simulate, render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import reducer from '../src/client/reducers';
import Main from '../src/client/components/Main.jsx';

const store = createStore(reducer);

test('Renders signup and signin forms', () => {
  const { getByText, container } =
    render(<Provider store={store}><Main /></Provider>);

  expect(container.firstChild).toMatchSnapshot();

  Simulate.click(getByText('Login'));
  expect(container.firstChild).toMatchSnapshot();

  Simulate.click(getByText('Sign up'));
  expect(container.firstChild).toMatchSnapshot();

  Simulate.click(getByText('Sign up'));
  expect(container.firstChild).toMatchSnapshot();
});
