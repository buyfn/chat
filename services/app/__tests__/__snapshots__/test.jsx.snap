// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders signup and signin forms 1`] = `
<div>
  <div
    class="userCounter"
  >
    Online: 
    0
  </div>
  <nav
    class="header"
  >
    <ul
      class="menu"
    >
      <li
        class="menu__item"
      >
        You are 
        anon
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Login
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Sign up
        </a>
      </li>
    </ul>
  </nav>
  <div
    style="display: none;"
  >
    <div
      class="loginForm"
      id="loginForm"
    >
      <form>
        <label
          class="loginForm__label"
        >
          Username: 
          <input
            id="loginUsername"
            name="loginUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="loginForm__label"
        >
          Password: 
          <input
            id="loginPassword"
            name="loginPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Login
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    style="display: none;"
  >
    <div
      class="signUpForm"
      id="signUpForm"
    >
      <form>
        <label
          class="signUpForm__label"
        >
          Username: 
          <input
            id="signupUsername"
            name="signupUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="signUpForm__label"
        >
          Password: 
          <input
            id="signupPassword"
            name="signupPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Sign up
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    class="chatForm"
  >
    <form>
      <input
        autocomplete="off"
        class="chatForm__input"
        id="newMessageText"
        name="newMessageText"
        required=""
        type="text"
        value=""
      />
      <button
        class="btn btn--main"
        id="sendMessage"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
  <div>
    <ul
      class="messageList"
    />
  </div>
</div>
`;

exports[`Renders signup and signin forms 2`] = `
<div>
  <div
    class="userCounter"
  >
    Online: 
    0
  </div>
  <nav
    class="header"
  >
    <ul
      class="menu"
    >
      <li
        class="menu__item"
      >
        You are 
        anon
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Login
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Sign up
        </a>
      </li>
    </ul>
  </nav>
  <div
    style="display: block;"
  >
    <div
      class="loginForm"
      id="loginForm"
    >
      <form>
        <label
          class="loginForm__label"
        >
          Username: 
          <input
            id="loginUsername"
            name="loginUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="loginForm__label"
        >
          Password: 
          <input
            id="loginPassword"
            name="loginPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Login
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    style="display: none;"
  >
    <div
      class="signUpForm"
      id="signUpForm"
    >
      <form>
        <label
          class="signUpForm__label"
        >
          Username: 
          <input
            id="signupUsername"
            name="signupUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="signUpForm__label"
        >
          Password: 
          <input
            id="signupPassword"
            name="signupPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Sign up
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    class="chatForm"
  >
    <form>
      <input
        autocomplete="off"
        class="chatForm__input"
        id="newMessageText"
        name="newMessageText"
        required=""
        type="text"
        value=""
      />
      <button
        class="btn btn--main"
        id="sendMessage"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
  <div>
    <ul
      class="messageList"
    />
  </div>
</div>
`;

exports[`Renders signup and signin forms 3`] = `
<div>
  <div
    class="userCounter"
  >
    Online: 
    0
  </div>
  <nav
    class="header"
  >
    <ul
      class="menu"
    >
      <li
        class="menu__item"
      >
        You are 
        anon
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Login
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Sign up
        </a>
      </li>
    </ul>
  </nav>
  <div
    style="display: none;"
  >
    <div
      class="loginForm"
      id="loginForm"
    >
      <form>
        <label
          class="loginForm__label"
        >
          Username: 
          <input
            id="loginUsername"
            name="loginUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="loginForm__label"
        >
          Password: 
          <input
            id="loginPassword"
            name="loginPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Login
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    style="display: block;"
  >
    <div
      class="signUpForm"
      id="signUpForm"
    >
      <form>
        <label
          class="signUpForm__label"
        >
          Username: 
          <input
            id="signupUsername"
            name="signupUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="signUpForm__label"
        >
          Password: 
          <input
            id="signupPassword"
            name="signupPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Sign up
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    class="chatForm"
  >
    <form>
      <input
        autocomplete="off"
        class="chatForm__input"
        id="newMessageText"
        name="newMessageText"
        required=""
        type="text"
        value=""
      />
      <button
        class="btn btn--main"
        id="sendMessage"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
  <div>
    <ul
      class="messageList"
    />
  </div>
</div>
`;

exports[`Renders signup and signin forms 4`] = `
<div>
  <div
    class="userCounter"
  >
    Online: 
    0
  </div>
  <nav
    class="header"
  >
    <ul
      class="menu"
    >
      <li
        class="menu__item"
      >
        You are 
        anon
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Login
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          class="pseudo"
          href="#"
        >
          Sign up
        </a>
      </li>
    </ul>
  </nav>
  <div
    style="display: none;"
  >
    <div
      class="loginForm"
      id="loginForm"
    >
      <form>
        <label
          class="loginForm__label"
        >
          Username: 
          <input
            id="loginUsername"
            name="loginUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="loginForm__label"
        >
          Password: 
          <input
            id="loginPassword"
            name="loginPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Login
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    style="display: none;"
  >
    <div
      class="signUpForm"
      id="signUpForm"
    >
      <form>
        <label
          class="signUpForm__label"
        >
          Username: 
          <input
            id="signupUsername"
            name="signupUsername"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          class="signUpForm__label"
        >
          Password: 
          <input
            id="signupPassword"
            name="signupPassword"
            required=""
            type="password"
            value=""
          />
        </label>
        <button
          class="btn btn--secondary"
          type="submit"
        >
          Sign up
        </button>
      </form>
      <span
        class="error"
      />
    </div>
  </div>
  <div
    class="chatForm"
  >
    <form>
      <input
        autocomplete="off"
        class="chatForm__input"
        id="newMessageText"
        name="newMessageText"
        required=""
        type="text"
        value=""
      />
      <button
        class="btn btn--main"
        id="sendMessage"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
  <div>
    <ul
      class="messageList"
    />
  </div>
</div>
`;
